lib/streamlit/storage/S3Storage.py

```diff
@@ -55,7 +55,7 @@ def __init__(self):
         log.propagate = False
 
         assert (
-            config.get_option("global.sharingMode") != "off"
+            config.get_option("global.sharingMode") == "s3"
```

Comment:
"s3" seems odd as the name for the "sharing mode" since it is more the name of the technology. Also, it is vendor dependent.

Reply:
Yeah - the s3 stuff is not actually part of the PR; this option existed before. It's possible that Streamlit can successfully share to other backends, but S3 is what the feature was originally designed around, and I think the only backend we've tested against.